daliuge-engine/dlg/deploy/slurm_client.py

```diff
@@ -55,47 +59,80 @@ class SlurmClient:
 
     def __init__(
         self,
-        dlg_root=None,
-        log_root=None,
-        host=None,
-        acc=None,
-        physical_graph_template_file=None,  # filename of physical graph template
-        logical_graph=None,
-        job_dur=30,
-        num_nodes=None,
-        run_proxy=False,
-        mon_host=DEFAULT_MON_HOST,
-        mon_port=DEFAULT_MON_PORT,
-        logv=1,
-        facility=None,
-        zerorun=False,
-        max_threads=0,
-        sleepncopy=False,
-        num_islands=None,
-        all_nics=False,
-        check_with_session=False,
-        submit=False,
-        remote=True,
-        pip_name=None,
-        username=None,
+        dlg_root: str = "",
```

Comment:
We have previously had a conversation about the issues with defaulting to empty string, as opposed to None. 

The reason I have replaced None here is that the type that is reported in the EAGLE component is "String", so when receive the graph JSON, all of the non-set parameters have the default `"'None"`. This means the following code:

```
if self.dlg_root:
    do_something()
```

Doesn't perform as expected, because the string `"None"` is not `None`. 

I'm open to alternatives on how to improve this.